### PR TITLE
Refactor gulp tests:integration

### PR DESCRIPTION
**Pull request by vxsx**
This PR starts the process of removing old selenium tests and replacing them with casperjs test

- refactors gulp tests:integration
     - starts the server from gulp
     - adds possibility to run different servers (with different options)
     - adds `--clean` option to remove the testdb before starting the server (cleanup crashed state when developing locally)
- documents changes (also documents `--screenshots`) 
- adds a tests for copying and pasting pages (for https://github.com/divio/django-cms/issues/5111) (which is mostly a copy paste from pagetree.js with small adaptions, so no need to review that one too hard)
- fixes small bug in cms helpers

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -171,95 +215,234 @@ gulp.task('tests:unit:watch', function () {
...
+
+    integrationTests
+        .prepareFiles()
+        .then(function (groupedTests) {
+            return Promise.reduce(Object.keys(groupedTests), function (items, serverArgs) {
+                var tests = groupedTests[serverArgs].map(function (obj) {
+                    return obj.file;
+                });
+                var serverPid;
+
```
Comment (FinalAngel) [inline gulpfile.js]: space

File: gulpfile.js [hunk 2]
```diff
@@ -19,6 +19,9 @@ var uglify = require('gulp-uglify');
 var KarmaServer = require('karma').Server;
 var child_process = require('child_process');
 var spawn = require('child_process').spawn;
+var terminate = require('terminate');
+var Promise = require('bluebird'); // jshint ignore:line
+var _ = require('lodash');
```
Comment (FinalAngel) [inline gulpfile.js]: is it common to call it `_`?

Comment (vxsx) [inline gulpfile.js]: yup

> // Load the full build.
> var _ = require('lodash');

https://lodash.com/

**Comment by FinalAngel** [comment]
Overall LGTM

considering the amount of tests, is it possible to categorise them further like:
- pagetree.persmissions.js
- pagetree.filter.js
- pagetree.copypastejs
- toolbar.login.js
- toolbar.logout.js
- toolbar.wizard.js
...

**Comment by vxsx** [comment]
@FinalAngel yeah, i was thinking about that, maybe use folders, maybe use namespaces

I'll do something, but not in this PR :)

**Comment by FinalAngel** [comment]
yeah no problem, just as a note :+1:

**Pull request merged**